code/scripting/api/libs/hud.cpp

```diff
@@ -237,6 +240,29 @@ ADE_FUNC(flashTargetBox, l_HUD, "enumeration section, [number duration_in_millis
 	return ADE_RETURN_NIL;
 }
 
+ADE_FUNC(getTargetDistance, l_HUD, "object targetee, [vector targeter_position]", "Returns the distance as displayed on the HUD, that is, the distance from a position to the bounding box of a target.  If targeter_position is nil, the function will use the player's position.", "number", "The distance, or nil if invalid")
+{
+	object_h *targetee;
+	vec3d *targeter_pos = nullptr;
+
+	if (!ade_get_args(L, "o|o", l_Object.GetPtr(&targetee), l_Vector.GetPtr(&targeter_pos)))
+		return ADE_RETURN_NIL;
+
+	if (targeter_pos == nullptr)
+	{
+		if (Player_obj)
+			targeter_pos = &Player_obj->pos;
+		else
+		{
+			Warning(LOCATION, "Player_obj is NULL; cannot assign position!");
+			targeter_pos = &vmd_zero_vector;
+		}
+	}
+
+	auto dist = hud_find_target_distance(targetee->objp, targeter_pos);
```

Comment:
`targetee` may be invalid so in order to avoid crashes, that should be checked before calling this.

Reply:
Good point.  Updated.

Reply:
You also need to check for an invalid handle since `objp` may be invalid otherwise.

Reply:
Missed that.  Updated again.